flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -196,8 +196,12 @@ public void open() {
 			if (line == null) {
 				continue;
 			}
-			final Optional<SqlCommandCall> cmdCall = parseCommand(line);
-			cmdCall.ifPresent(this::callCommand);
+			try {
+				final Optional<SqlCommandCall> cmdCall = parseCommand(line);
+				cmdCall.ifPresent(this::callCommand);
+			} catch (Throwable t) {
```

Comment:
We should not catch everything and let severe errors through. The problem lies deeper in the `Executor` that should wrap all exceptions into `SqlExecutionException`.

Reply:
I think SQL client crashed is a terrible thing even a small problem occurs. For example: before [FLINK-16539](https://issues.apache.org/jira/browse/FLINK-16539) is fixed,  the client will crash even users execute this simple command: `set execution.parallelism = 10;` (` 10` can't be converted to Integer because there is a blank before `10`). We just need to print the error message to let users know where is the problem, and then users can continue to execute the modified command. Even we wrap all exceptions in the `Executor` into `SqlExecutionException`, there may be other bugs hidden.

Reply:
@twalthr have a look if you have time ?

Reply:
+1 to catch everything here. Even we are able to wrap all exceptions into `SqlExecutionException`, there still would be some unexpected errors, like NPE or something like that. There is no real harm for catching everything here, users still can see the error messages from console, knowing something went wrong.

Reply:
The SQL Client process should not continue if something unexpected happened. The process might be in a bad state then. We should fail hard in those cases. The current exception design is also considering this. In the main() method a wrapping message will be displayed that says `This is a bug. Please file an issue.`. We should rather fix `CliClient#callSet()` in this case instead of messing with the exception design.

Reply:
> The process might be in a bad state then

This makes sense, there is a risk that we will do more damage if don't exist.

Reply:
@twalthr `catch Throwable` is removed. instead, I catch ValidationException and throw SqlExecutionException when enriching properties, and check all `callXX` methods in CliClient to make sure all SqlExecutionExceptions are caught.